Домашнее задание по спринту 8

Kris-found opened the pull request.

Kris-found committed:
Домашнее задание по спринту 8

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,23 @@
+
+        arrowBackButtonInSettings.setOnClickListener{
+            val arrowBackInMain = Intent(this, MainActivity::class.java)
+            startActivity(arrowBackInMain)
```
Для перехода назад надо использовать не интент, а методы Activity finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити.

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,76 @@
+            app:iconGravity="textStart"
+            app:iconPadding="@dimen/icon_padding"/>
+
+</LinearLayout>
```
Поплыло форматирование в файле. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране. Советую прожимать каждый раз после окончания работы в файле

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,152 @@
+            android:textColor="@color/black"
+            android:textStyle="bold"
+            android:layout_gravity="center_vertical"
+            android:layout_marginStart="64dp"/>
```
Не стоит в таких случаях задавать взаимное расположение элементов большим отступом от начала экрана. Это моветон. Тут напрашивается горизонтальный LinearLayout вместо FrameLayout, где сперва идет TextView, а потом ImageView.

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,10 @@
+    <dimen name="icon_padding">8dp</dimen>
+    <dimen name="button_radius">16dp</dimen>
+    <dimen name="settings_text_size">16sp</dimen>
+    <dimen name="settings_lists_padding">21dp</dimen>
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). По крайней мере чётными. Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Лучше по возможности этой рекомендации следовать. Хоть дизайнеры не всегда следуют гайдлайнам гугла, наша задача сделать такой UI, чтобы он соответствовал макетам, а не чтобы числа были одинаковые. Тем более что всем размерам из фигмы угодить невозможно невозможно, учитывая огромное количество форм и размеров устройств. Так что если мы сделаем и UI как на макетах, и при этом последуем лучшим практикам, то будет круто). Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

Kris-found committed:
Задачи на 9 спринт. Реализованы:
1. Темы и стили
2. Экран "Настройки" с логикой
3. Экран "Поиска" - окно поиска, логика кнопки "Очистить", поддержка изменения ориенитации экрана без потери вводимых данных
